lib/connectors_sdk/office365/extractor.rb

```diff
@@ -61,14 +51,17 @@ def yield_document_changes(modified_since: nil, break_after_page: false, &block)
             capture_exception(e)
           end
 
-          if break_after_page
-            if config.cursors['page_cursor'].blank?
-              config.cursors['last_drive_id'] = config.cursors.delete('current_drive_id')
-            end
+          if break_after_page && config.cursors['page_cursor'].present?
             break
           end
         end
 
+        if break_after_page && config.cursors['page_cursor'].blank?
+          @completed = true
+          config.overwrite_cursors!(retrieve_latest_cursors)
+          log_info("Completed #{modified_since.nil? ? 'full' : 'incremental'} extraction")
```

Comment:
this just gave me a light-bulb moment - should we just add DEBUG logs for everywhere we're modififying and setting the `cursors`? Maybe that would help us to troubleshoot in the future, and feel good about our changes now.